### PR TITLE
Refactoring and fix

### PR DESCRIPTION
**Pull request by phunkyfish**
v3.12.2
- Fixed: Refactoring - Conventions: includes, namespace naming, public private order in class definition
- Fixed: GetInitialEPGForGroup called for each CHANNEL while initial EPG Update - #86

### CODE REVIEW

**Comment by phunkyfish** [comment]
Ok, so the first commit is a start on bringing the base in line with conventions. Next I would like to move the files to a structure under the root and split of the classes/structs into their own files #100. 

On trying to do this classes must start with a C, is there a convention for structs or do you tend to leave them with classes and not put them in their own files?

When I move the files to a new directory structure it will all appear like new code. I will still have more cleanup to do and would rather not have to do it all in one go. Is approval possible just on the file move and split not worrying about the functional content?

**Comment by phunkyfish** [comment]
On the second commit it's a fix which should bring some performance improvements for users. There were a few issues but moving to a myGroupPtr appears to solve it. I'm not delighted with it but it will do till I get around to #101 below.

During the investigation I discovered that a channel is create for occurrence of a channel in a group. Meaning channels exist multiple times. #101 would help resolve this and give a nice performance boost to users. It's an extensive change though.

**Comment by Jalle19** [comment]
@phunkyfish the C prefix is used in Kodi, but addons can folllw their own rules. Pvr.hts fpr example has namespaces and non-prefixed classes.

**Comment by phunkyfish** [comment]
Good to know. Thanks!

**Comment by phunkyfish** [comment]
It appears I forgot to add the version and changelog adding now.

File: src/VuData.cpp [hunk 1]
```diff
@@ -1245,9 +1269,24 @@ PVR_ERROR Vu::GetInitialEPGForChannel(ADDON_HANDLE handle, const VuChannel &chan
       PVR->TransferEpgEntry(handle, &broadcast);
     }
   }
+
   return PVR_ERROR_NO_ERROR;
 }
 
+bool Vu::CheckIfAllChannelsHaveInitialEPG() const
```
Comment (ksooo) [inline src/VuData.cpp]: This function could be static as it does. not operate on any members of Vu class.

File: src/VuData.cpp [hunk 2]
```diff
@@ -1260,35 +1299,30 @@ PVR_ERROR Vu::GetEPGForChannel(ADDON_HANDLE handle, const PVR_CHANNEL &channel,
...
 
-  VuChannel myChannel;
-  myChannel = m_channels.at(channel.iUniqueId-1);
+  VuChannel& myChannel = m_channels.at(channel.iUniqueId-1);
+
+  XBMC->Log(LOG_DEBUG, "%s Getting EPG for channel '%s'", __FUNCTION__, myChannel.strChannelName.c_str());
 
   // Check if the initial short import has already been done for this channel
-  if (m_channels.at(channel.iUniqueId-1).bInitialEPG == true)
+  if (myChannel.bRequiresInitialEPG == true)
```
Comment (ksooo) [inline src/VuData.cpp]: There's no need for the `== true`.

**Comment by phunkyfish** [comment]
Ok, should be good.

**Pull request merged**